ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1669,25 +1657,7 @@ FEDataManager::interpWeighted(const int f_data_idx,
                 Elem* const elem = patch_elems[e_idx];
                 const quad_key_type& key = quad_keys[e_idx];
                 QBase& qrule = d_quadrature_cache[key];
-                FEBase& X_fe = X_fe_cache[key];
-
-                // libMesh::FE defaults to recalculating *everything* when we
-                // call reinit unless we first request things (see
-                // FEGenericBase<>::determine_calculations() for the
-                // reinitialization logic). This only works for Lagrange
-                // elements (and similar nodal value elements).
-                //
-                // Furthermore: we can skip calling reinit() completely as
-                // long as we have called it on this FE object at least once.
-                if (used_X_quadratures.find(key) == used_X_quadratures.end())
-                {
-                    // get_phi changes the state of the object if called
-                    // before reinit
-                    X_fe.get_phi();
-                    X_fe.reinit(elem);
-                    used_X_quadratures.insert(key);
-                }
-
```

Comment:
A nice thing about this patch is that I got to delete a previous round of hacks, now that I know how to implement `FECache` correctly.